cpp/util/etcd.cc

```diff
@@ -573,9 +573,8 @@ void EtcdClient::WatchRequestDone(WatchState* state, GetResponse* get_resp,
     return;
   }
 
-  // This is probably due to a timeout, just retry.
-  if (!child_task->status().ok()) {
-    LOG_EVERY_N(INFO, 10) << "Watch request failed: " << child_task->status();
+  if (child_task->status().error_code() == util::error::DEADLINE_EXCEEDED) {
+    VLOG(1) << "Watch request timedout: " << child_task->status();
```

Comment:
I think for watches, we want to keep them trying forever, unless there was some sort of error that could be truly final?

For example, all etcd might go offline for a bit, nothing much is going to happen, but we should keep trying, and resume notifying once they come back, I'd say.